### ISSUE TITLE
Add file nesting

### ISSUE THREAD

**Issue by jacobdalamb** [zed-industries/zed#7092]
### Check for existing issues

- [X] Completed

### Describe the feature

A setting that enables users to create "folders" for specific files such as configuration files for frameworks, build and test tools and others. This idea is from a VSCode extension [file-nesting](https://marketplace.visualstudio.com/items?itemName=antfu.file-nesting) that creates a config snippet using VS Code [file nesting feature](https://code.visualstudio.com/updates/v1_67#_explorer-file-nesting)

### If applicable, add mockups / screenshots to help present your vision of the feature

![image](https://github.com/zed-industries/zed/assets/44789941/ca0782a4-bbdf-4039-9ef7-03491b4d84b9)

**Comment by Hidde-Heijnen** [comment]
would love to see this!

**Comment by saifulapm** [comment]
Yes.. This will be awesome

**Comment by thoroc** [comment]
Especially useful when the test files are named `<filename>.(test|spec).<extension>` and placed next to the source file, which happens a lot in the JS/TS world.

**Comment by bcardarella** [comment]
If this is going to happen could there be configurable patterns so different frameworks could ship their nesting patterns?

**Comment by LANIK2000** [comment]
Yes please! I'd especially appreciate if we could nest rust files too. So the `./my_mod.rs` file and and `./my_mod/` directory. Surprisingly haven't seen this anywhere else yet, or am to dumb to find it.

**Comment by bcardarella** [comment]
@ConradIrwin you marked as "completed" but from your comment this seems like a not planned as you implied it should be through an extension?

**Comment by ConradIrwin** [comment]
Hmm, I think that's just the GitHub close button behavior - definitely not completed, agreed

**Comment by tkgalk** [comment]
Why was this closed with no follow-up. There is currently no support for extensions to do so, so it should remain as an open issue/feature request (whether it gets picked up is something else). When extensions can do this, sure, but until then it feels a bit harsh.
